src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Typeface.cs

```diff
@@ -632,7 +642,7 @@ out idealWidth
             GlyphTypeface glyphTypeface = TryGetGlyphTypeface();
             Invariant.Assert(glyphTypeface != null);
 
-            MS.Internal.Text.TextInterface.GlyphMetrics[] glyphMetrics = BufferCache.GetGlyphMetrics(charBufferRange.Length);
+            MS.Internal.Text.TextInterface.GlyphMetrics[] glyphMetrics = ArrayPool<MS.Internal.Text.TextInterface.GlyphMetrics>.Shared.Rent(charBufferRange.Length);
```

Comment:
Thank you for your contribution. Sorry, I don't believe that placing GlyphMetrics into the ArrayPool is a genuine point of optimization. The reason is that GlyphMetrics is an array type that is only used in this context, and thus it cannot be reused with other modules through ArrayPool. In order to create a shared GlyphMetrics array, it would be necessary to create additional objects such as `SharedArrayPool<GlyphMetrics>`, which seems to have a higher cost than benefit. More importantly, the GlyphMetrics array is an object that is frequently used and then released. It can be assumed that virtually all modern WPF applications have text, meaning they constantly require the acquisition and return of GlyphMetrics arrays. In this case, it might be worth considering using a local variable to cache it. Perhaps the cost of using a local variable for caching is the lowest, and the benefit is the highest.

![image](https://github.com/dotnet/wpf/assets/16054566/927819fd-8dea-4ec8-82ad-d5decce3ec7a)

Reply:
Ok, let me group proposed solutions:
1. `GlyphMetrics[]` - approach similar to `BufferCache` without automatic release
2. `ConditionalWeakTable` - automatic release of `ArrayPool`? SharedArrayPool already has a mechanism to detect unused arrays and free them on [garbage collection event](https://github.com/dotnet/runtime/blob/6ebc8bd86dbc780b2a2a7daf3ab6020f9104f09e/src/libraries/System.Private.CoreLib/src/System/Buffers/SharedArrayPool.cs#L291)
3. `SharedArrayPool` - proposed solution

Also, perhaps we could use `stackalloc` array here if the number of elements is not too high (maybe 150 (for 4096 / `sizeof(GlyphMetrics)`))